server/devdocs/bin/generate-proptypes-index.js

```diff
@@ -4,31 +4,99 @@
  * This file generates an index of proptypes by component displayname, slug and folder name
  */
 
+const startTime = process.hrtime();
+
 /**
  * External Dependencies
  */
-
+require( 'babel-register' );
 const fs = require( 'fs' );
 const path = require( 'path' );
 const reactDocgen = require( 'react-docgen' );
+const { getPropertyName, getMemberValuePath, resolveToValue } = require( 'react-docgen/dist/utils' );
+const util = require( 'client/devdocs/docs-example/util' );
 
 const root = path.dirname( path.join( __dirname, '..', '..' ) );
-const pathSwap = new RegExp(path.sep, 'g');
+const pathSwap = new RegExp( path.sep, 'g' );
+const handlers = [ ...reactDocgen.defaultHandlers, commentHandler ];
 
 /**
- * Converts a camel cased string into a slug
- * @param {String} name The camel cased string to slugify
- * @return {String}
+ * Replaces **'s in comment blocks and trims comments
+ * @param {string} str The doc string to clean
+ * @return {string} The clean comment block text
  */
-const camelCaseToSlug = ( name ) => {
-	if ( ! name ) {
-		return name;
+function parseDocblock( str ) {
+	const lines = str.split( '\n' );
+	for ( let i = 0, l = lines.length; i < l; i++ ) {
+		lines[ i ] = lines[ i ].replace( /^\s*\*\s?/, '' );
 	}
+	return lines.join( '\n' ).trim();
+}
 
-	return name
-		.replace( /\.?([A-Z])/g, ( x, y ) => '-' + y.toLowerCase() )
-		.replace( /^-/, '' );
-};
+/**
+ * Given a path, this function returns the closest preceding comment, if it exists
+ * @param {NodePath} nodePath The node path from react-docgen
+ * @return {null|string} The comment or nothing, if no comment
+ */
+function getComments( nodePath ) {
```

Comment:
Since this file is self-executing, based off of `argv` there's three ways I can think to make it testable:

1. Refactor the file into modules that are imported; test the modules -- lots of granularity
2. execute the script from the test -- more of an integration test, really
3. all of the above

I'm partial to "all of the above" ... what are your thoughts?